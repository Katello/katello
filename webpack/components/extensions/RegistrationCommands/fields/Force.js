import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'foremanReact/common/helpers';

import { FormGroup, Checkbox } from '@patternfly/react-core';
import LabelIcon from 'foremanReact/components/common/LabelIcon';
import { translate as __ } from 'foremanReact/common/I18n';

const Force = ({ value, onChange, isLoading }) => (
  <FormGroup fieldId="reg_katello_force">
    <Checkbox
      label={
        <span>
          {__('Force')}{' '}
          <LabelIcon text={__('Remove any `katello-ca-consumer` rpms before registration and run subscription-manager with `--force` argument.')} />
        </span>
      }
      id="reg_katello_force"
      onChange={() => onChange({ force: !value })}
      isDisabled={isLoading}
      isChecked={value}
    />
  </FormGroup>
);

Force.propTypes = {
  value: PropTypes.bool,
  onChange: PropTypes.func,
  isLoading: PropTypes.bool,
};

Force.defaultProps = {
  value: false,
  onChange: noop,
  isLoading: false,
};

export default Force;
