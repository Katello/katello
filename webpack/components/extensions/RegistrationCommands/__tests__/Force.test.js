import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';
import Force from '../fields/Force';

const fixtures = {
  renders: { value: false, isLoading: false, onChange: () => {} },
};

describe('Force', () =>
  testComponentSnapshotsWithFixtures(Force, fixtures));
