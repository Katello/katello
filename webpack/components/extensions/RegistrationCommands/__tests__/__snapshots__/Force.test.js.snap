// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Force renders 1`] = `
<FormGroup
  fieldId="reg_katello_force"
>
  <Checkbox
    className=""
    id="reg_katello_force"
    isChecked={false}
    isDisabled={false}
    isLabelWrapped={false}
    isRequired={false}
    isValid={true}
    label={
      <span>
        Force
         
        <LabelIcon
          text="Remove any \`katello-ca-consumer\` rpms before registration and run subscription-manager with \`--force\` argument."
        />
      </span>
    }
    onChange={[Function]}
    ouiaId="reg-katello-force"
    ouiaSafe={true}
  />
</FormGroup>
`;
